sentry/src/main/java/io/sentry/SentryTracer.java

```diff
@@ -366,10 +365,6 @@ public void finish(@Nullable SpanStatus status) {
       }
 
       transaction.getMeasurements().putAll(measurements);
-
-      if (profilingTraceData != null) {
```

Comment:
there was no test for this previously? Sounds like it should fail

Reply:
previously it was captured with the transaction, and the only required parameter was the transaction (the others could be null)
then I switched to send the profile directly using `hub.captureProfile` and added the null check
now i moved the `captureProfile` call into the Profiler directly
the test check if finishing a transaction calls the `profiler.onTransactionFinish` method, and the profiler tests check if the `ISentryClient.captureProfile` is called